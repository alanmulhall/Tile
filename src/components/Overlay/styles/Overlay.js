import styled from 'styled-components';

export const OverlayStyled = styled.div`
  background: rgba(0, 0, 0, 0.65);
  box-sizing: border-box;
  color: white;
  left: 0;
  bottom: 0;
  padding: ${props => props.theme.spacing.xm} ${props => props.theme.spacing.m};
  position: absolute;
  height: ${props => props.fullscreen ? '100%' : 'auto'};
  width: 100%;
  z-index: ${props => props.theme.zIndex[0]};
`;
