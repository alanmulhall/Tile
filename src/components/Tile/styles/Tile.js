import styled from 'styled-components';
import SevenMark from '../../SevenMark/SevenMark';
import ResponsiveImage from '../../ResponsiveImage/ResponsiveImage';

const sizes = {
  small: 1,
  medium: 1.5,
  large: 2
};

export const Caption = styled.figcaption`
  align-items: center;
  display: flex;
`;

export const Image = styled(ResponsiveImage)`
  &:hover {
    transform: scale(1.1);
  }
`;

export const TileStyled = styled.figure`
  cursor: pointer;
  height: calc(177px * ${props => sizes[props.size]});
  position: relative;
  overflow: hidden;
  width: calc(315px * ${props => sizes[props.size]});
`;

export const Logo = styled(SevenMark)`
  height: 34px;
  margin-right: ${props => props.theme.spacing.m};
`;
