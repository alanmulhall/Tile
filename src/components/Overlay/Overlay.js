import React from 'react';
import PropTypes from 'prop-types';
import { OverlayStyled } from './styles/Overlay';

const defaultProps = {
  children: null,
  className: null,
  fullscreen: true
};

const propTypes = {
  children: PropTypes.node,
  className: PropTypes.string,
  fullscreen: PropTypes.bool
};

const Overlay = ({ children, className, fullscreen }) => (
  <OverlayStyled className={className} fullscreen={fullscreen}>
    {children}
  </OverlayStyled>
);

Overlay.defaultProps = defaultProps;
Overlay.propTypes = propTypes;

export default Overlay;
