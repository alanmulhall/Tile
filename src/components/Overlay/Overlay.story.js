import React from 'react';
import { storiesOf } from '@storybook/react';
import { boolean, withKnobs } from '@storybook/addon-knobs';
import centered from '@storybook/addon-centered';
import withTests from '../../utils/withTests';
import Overlay from './Overlay';

const OverlayStory = () =>
  storiesOf('Overlay', module)
    .addDecorator(centered)
    .addDecorator(withKnobs)
    .addDecorator(withTests('Overlay'))
    .add('Default', () => <Overlay fullscreen={boolean('Fullscreen', true)} />);

export default OverlayStory;
