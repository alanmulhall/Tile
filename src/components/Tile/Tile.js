import React, { PureComponent } from 'react';
import PropTypes from 'prop-types';
import { Caption, Image, Logo, TileStyled } from './styles/Tile';
import Title from '../../components/H2/H2';
import Overlay from '../Overlay/Overlay';

class Tile extends PureComponent {
  static defaultProps = {
    size: 'small'
  };

  static propTypes = {
    image: PropTypes.string.isRequired,
    onClick: PropTypes.func.isRequired,
    size: PropTypes.oneOf(['small', 'medium', 'large']),
    title: PropTypes.string.isRequired
  };

  render() {
    const { image, onClick, size, title } = this.props;
    return (
      <TileStyled onClick={onClick} size={size}>
        <Image src={image} alt={title} />
        <Overlay fullscreen={false}>
          <Caption>
            <Logo />
            <Title>{title}</Title>
          </Caption>
        </Overlay>
      </TileStyled>
    );
  }
}

export default Tile;
