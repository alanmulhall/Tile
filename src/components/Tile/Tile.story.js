import React from 'react';
import { storiesOf } from '@storybook/react';
import centered from '@storybook/addon-centered';
import withTests from '../../utils/withTests';
import Tile from './Tile';

const TileStory = () =>
  storiesOf('Tile', module)
    .addDecorator(centered)
    .addDecorator(withTests('Tile'))
    .add('Default', () => <Tile />);

export default TileStory;
