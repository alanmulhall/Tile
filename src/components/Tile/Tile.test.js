import React from 'react';
import 'jest-styled-components';
import renderer from '../../utils/rendererThemeProviderWrapper.js';
import shallow from '../../utils/shallowThemeProviderWrapper';
import mount from '../../utils/mountThemeProviderWrapper';
import Tile from './Tile';
import Overlay from '../Overlay/Overlay';
import { Caption, Image } from './styles/Tile';
import Title from '../H2/H2';
import tileImage from '../../images/tile.jpg';

describe('<Tile image={tileImage} />', () => {
  it('renders successfully', () => {
    const tree = renderer(<Tile image={tileImage} title="Home and Away" onClick={() => {}} />);
    expect(tree).toMatchSnapshot();
  });

  it('calls "onClick" handler', () => {
    const spy = jest.fn();
    const wrapper = shallow(<Tile image={tileImage} title="Home and Away" onClick={spy} />);
    wrapper.simulate('click');
    expect(spy).toHaveBeenCalled();
  });

  it('has a <Image /> component', () => {
    const wrapper = mount(<Tile image={tileImage} title="Home and Away" onClick={() => {}} />);
    expect(wrapper.find(Image).length).toEqual(1);
  });

  it('has a <Overlay /> component', () => {
    const wrapper = mount(<Tile image={tileImage} title="Home and Away" onClick={() => {}} />);
    expect(wrapper.find(Overlay).length).toEqual(1);
  });

  it('has a <Caption /> component', () => {
    const wrapper = mount(<Tile image={tileImage} title="Home and Away" onClick={() => {}} />);
    expect(wrapper.find(Caption).length).toEqual(1);
  });

  it('has a <Title /> component', () => {
    const wrapper = mount(<Tile image={tileImage} title="Home and Away" onClick={() => {}} />);
    expect(wrapper.find(Title).length).toEqual(1);
  });

  describe('size', () => {
    describe('small', () => {
      it('renders successfully', () => {
        const tree = renderer(<Tile image={tileImage} size="small" title="Home and Away" onClick={() => {}} />);
        expect(tree).toMatchSnapshot();
      });
    });

    describe('medium', () => {
      it('renders successfully', () => {
        const tree = renderer(<Tile image={tileImage} size="medium" title="Home and Away" onClick={() => {}} />);
        expect(tree).toMatchSnapshot();
      });
    });

    describe('large', () => {
      it('renders successfully', () => {
        const tree = renderer(<Tile image={tileImage} size="large" title="Home and Away" onClick={() => {}} />);
        expect(tree).toMatchSnapshot();
      });
    });
  });
});
