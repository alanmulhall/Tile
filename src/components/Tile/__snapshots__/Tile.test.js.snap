// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Tile /> renders successfullly 1`] = `
.c0 {
  color: grey;
}

<div
  className="c0"
>
  Tile
</div>
`;
